iroha_permissions_validators/src/lib.rs

```diff
@@ -16,189 +17,277 @@ pub mod public_blockchain {
+                    .read_asset_definition_entry(&asset_id.definition_id)
+                    .map_or(false, |asset_definiton_entry| {
+                        asset_definiton_entry.registered_by != authority
+                    });
+                if low_authority {
+                    return Err("Can't mint assets registered by other accounts.".to_owned());
                 }
+                Ok(())
             }
-            Ok(())
         }
-    }
 
-    impl From<BurnOnlyAssetsCreatedByThisAccount> for PermissionsValidatorBox {
-        fn from(_: BurnOnlyAssetsCreatedByThisAccount) -> Self {
-            Box::new(BurnOnlyAssetsCreatedByThisAccount)
-        }
-    }
+        /// Checks that account can burn only the assets that he currently owns.
+        #[derive(Debug, Copy, Clone)]
+        pub struct OnlyOwnedAssets;
 
-    /// Checks that account can burn only the assets that he currently owns.
-    #[derive(Debug, Copy, Clone)]
-    pub struct BurnOnlyOwnedAssets;
-
-    impl PermissionsValidator for BurnOnlyOwnedAssets {
-        fn check_instruction(
-            &self,
-            authority: AccountId,
-            instruction: Instruction,
-            wsv: &WorldStateView,
-        ) -> Result<(), DenialReason> {
-            if let Instruction::Burn(instruction) = instruction {
-                if let IdBox::AssetId(asset_id) = instruction
+        from_boxed_unit!(OnlyOwnedAssets);
+
+        impl PermissionsValidator for OnlyOwnedAssets {
+            fn check_instruction(
+                &self,
+                authority: AccountId,
+                instruction: Instruction,
+                wsv: &WorldStateView,
+            ) -> Result<(), DenialReason> {
+                let instruction: BurnBox = try_into_or_exit!(instruction);
+                let instruction = instruction
                     .destination_id
                     .evaluate(wsv, &Context::new())
-                    .map_err(|e| e.to_string())?
-                {
-                    if asset_id.account_id != authority {
-                        return Err("Can't burn assets from another account.".to_owned());
-                    }
+                    .map_err(|e| e.to_string())?;
+                let asset_id: AssetId = try_into_or_exit!(instruction);
+                if asset_id.account_id != authority {
+                    return Err("Can't burn assets from another account.".to_owned());
                 }
+                Ok(())
             }
-            Ok(())
         }
     }
 
-    impl From<BurnOnlyOwnedAssets> for PermissionsValidatorBox {
-        fn from(_: BurnOnlyOwnedAssets) -> Self {
-            Box::new(BurnOnlyOwnedAssets)
+    pub mod keyvalue {
+        //! Module with permission for burning
+
+        use super::*;
+
+        /// Checks that account can burn only the assets which were registered by this account.
+        #[derive(Debug, Copy, Clone)]
+        pub struct SetOnlyCreatedByThisAccount;
```

Comment:
Conceptually for key-value assets I think it will make most sense that, only asset owner can set its values, not the asset definition creator. Same for remove.

Example:
1. Alice creates AssetDefinition Diary
2. Bob adds a first entry, therefore creating this asset at his account - Asset(Diary)
3. Clair adds a first entry, therefore creating this asset at her account - Asset(Diary)
4. Clair tries to add an entry to Bob's Asset(Diary) - this is not permitted
5. Clair adds a second entry to her own Asset(Diary)
6. Clair removes the first entry from her own Asset(Diary)
7. ...

Also please correct the comments, I think they were copied and not updated.

Reply:
Oh, you actually implemented it correctly, it is just the confusing name. Please rename to `SetOnlyForSignerAccount`, meaning the signer of tx will only be able to set to asset store and metadata to his assets and account.

Reply:
Also I suggest to split these checks into two - setting of account metadata and setting of asset store, same for remove. We are trying to make permissions very granular, so that each project can configure their specific set with the use of some of hours.